app/javascript/packages/captcha-submit-button/captcha-submit-button-element.ts

```diff
@@ -0,0 +1,78 @@
+export const CAPTCHA_EVENT_NAME = 'lg:captcha-submit-button:challenge';
+
+class CaptchaSubmitButtonElement extends HTMLElement {
+  connectedCallback() {
+    this.button.addEventListener('click', (event) => this.handleButtonClick(event));
+  }
+
+  get button(): HTMLButtonElement {
+    return this.querySelector('button')!;
+  }
+
+  get tokenInput(): HTMLInputElement {
+    return this.querySelector('[type=hidden]')!;
+  }
+
+  get form(): HTMLFormElement | null {
+    return this.closest('form');
+  }
+
+  get recaptchaSiteKey(): string | null {
+    return this.getAttribute('recaptcha-site-key');
+  }
+
+  get recaptchaAction(): string {
+    return this.getAttribute('recaptcha-action')!;
+  }
+
+  submit() {
+    this.form?.submit();
+  }
+
+  invokeChallenge() {
+    grecaptcha.ready(async () => {
+      const { recaptchaSiteKey: siteKey, recaptchaAction: action } = this;
+      const token = await grecaptcha.execute(siteKey!, { action });
```

Comment:
In production, if we have an enterprise account, we may want to call:

grecaptcha.enterprise.execute
grecaptcha.enterprise.ready

https://cloud.google.com/recaptcha-enterprise/docs/

Reply:
Yeah, I wondered about this too, since I saw that there was a separate namespace for the enterprise functions. The migration guide you referenced led me to believe that it's not strictly necessary?

>The migration process takes 5–10 minutes to complete and requires no code changes.

It's difficult to confirm before having access to credentials. I can create a ticket to track this work as part of the enterprise migration, since we'll also want to consider if we'll still need to also support the free version (e.g. to facilitate local testing, etc.).

Reply:
Since [the other guide](https://cloud.google.com/recaptcha-enterprise/docs/using-features) you shared mentions these being relevant for features specific to reCAPTCHA Enterprise, we'll probably want to consider migrating eventually anyways. I'll also plan to incorporate the backend call endpoint into the existing work for LG-8826, and write a new ticket for other enterprise-related migration work.

Reply:
I created a ticket LG-8891 for the enterprise migration. I've also been working with TTS Tech Portfolio to get access to Google Cloud Platform for generating credentials.